source/apiVolontaria/volunteer/tests/tests_view_CyclesId.py

```diff
@@ -0,0 +1,181 @@
+from rest_framework.test import APIClient
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from volunteer.models import Cycle
+from apiVolontaria.factories import UserFactory, AdminFactory
+from django.utils import timezone
+import json
+
+
+class CyclesIdTests(APITestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+        self.user = UserFactory()
+        self.user.set_password('Test123!')
+        self.user.save()
+
+        self.admin = AdminFactory()
+        self.admin.set_password('Test123!')
+        self.admin.save()
+
+        self.cycle = Cycle.objects.create(
+            name='Cycle 1',
+            start_date=timezone.now(),
+            end_date=timezone.now() + timezone.timedelta(
+                minutes=100,
+            ),
+        )
+
+    def test_retrieve_cycle_id_not_exist(self):
+        """
+        Ensure we can't retrieve a cycle who doesn't exist.
```

Comment:
Ensure we can't retrieve a cycle ~who~ **that** doesn't exist. 

"Who" should be used when speaking of people, but not ojects (such as cycles).